apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.js

```diff
@@ -81,256 +95,183 @@ router.post(
+      questionTestPath += '/';
+    }
+    questionTestPath += 'test';
 
-  // Generate a CSRF token for the test route. We can't use `res.locals.__csrf_token`
-  // here because this form will actually post to a different route, not `req.originalUrl`.
-  const questionTestCsrfToken = generateSignedToken(
-    { url: questionTestPath, authn_user_id: res.locals.authn_user.user_id },
-    config.secretKey,
-  );
+    // Generate a CSRF token for the test route. We can't use `res.locals.__csrf_token`
+    // here because this form will actually post to a different route, not `req.originalUrl`.
+    const questionTestCsrfToken = generateSignedToken(
+      { url: questionTestPath, authn_user_id: res.locals.authn_user.user_id },
+      config.secretKey,
+    );
 
-  res.locals.questionTestPath = questionTestPath;
-  res.locals.questionTestCsrfToken = questionTestCsrfToken;
+    res.locals.questionTestPath = questionTestPath;
+    res.locals.questionTestCsrfToken = questionTestCsrfToken;
 
-  res.locals.isEnterprise = isEnterprise();
+    res.locals.isEnterprise = isEnterprise();
 
-  async.series(
-    [
-      (callback) => {
-        res.locals.questionGHLink = null;
-        if (res.locals.course.repository) {
-          const GHfound = res.locals.course.repository.match(
-            /^git@github.com:\/?(.+?)(\.git)?\/?$/,
-          );
-          if (GHfound) {
-            res.locals.questionGHLink =
-              'https://github.com/' +
-              GHfound[1] +
-              '/tree/master/questions/' +
-              res.locals.question.qid;
-          }
-        } else if (res.locals.course.example_course) {
-          res.locals.questionGHLink = `https://github.com/PrairieLearn/PrairieLearn/tree/master/exampleCourse/questions/${res.locals.question.qid}`;
-        }
-        callback(null);
-      },
-      (callback) => {
-        sqldb.queryOneRow(sql.qids, { course_id: res.locals.course.id }, (err, result) => {
-          if (ERR(err, callback)) return;
-          res.locals.qids = result.rows[0].qids;
-          callback(null);
-        });
-      },
-      (callback) => {
-        sqldb.query(
-          sql.select_assessments_with_question_for_display,
-          { question_id: res.locals.question.id },
-          (err, result) => {
-            if (ERR(err, callback)) return;
-            res.locals.a_with_q_for_all_ci = result.rows[0].assessments_from_question_id;
-            callback(null);
-          },
-        );
-      },
-      async () => {
-        res.locals.sharing_enabled = await features.enabledFromLocals(
-          'question-sharing',
-          res.locals,
-        );
+    res.locals.questionGHLink = null;
+    if (res.locals.course.repository) {
+      const GHfound = res.locals.course.repository.match(/^git@github.com:\/?(.+?)(\.git)?\/?$/);
+      if (GHfound) {
+        res.locals.questionGHLink =
+          'https://github.com/' + GHfound[1] + '/tree/master/questions/' + res.locals.question.qid;
```

Comment:
Consider updating this to support branch names other than `master`. We already store this in the `pl_courses.branch` column, which you should be able to get at via `res.locals.course.branch`.

Reply:
Updated to use `res.locals.course.branch` instead of hard-coding "master" for the branch name.